### ISSUE TITLE
Change lib files without compiling the entire project again

### ISSUE THREAD

**Issue by exx8** [nodejs/help#2228]
* **Node.js Version**:
12.11
* **OS**:
ubuntu 18.04
* **Scope (install, code, runtime, meta, other?)**:
debugging

I'm trying to contribute to the project via coding.
The build process takes a lot of time. I would be happy to know:

1.  if there is a way to change lib files without rebuilding the entire project?
2. Is there are a debugger for the lib files themselves?

Thanks

**Comment by addaleax** [comment]
> if there is a way to change lib files without rebuilding the entire project?

Running `make` should only re-compile the parts of the project that have been changed and files that depend on those, so it shouldn’t re-build the entire project. (“should” because our build system has been broken for a while now, and *some* parts are always re-compiled and it’s not bothering anybody enough to do something about it).

Another way to keep compile times down is to install `ccache`, which makes sure that re-compiling will use a cache even when changing branches a lot.

As for being able to modify `lib/` files without re-compiling Node.js, that’s currently not possible, no. (I tried to change that ~ 2 years ago but got a lot of pushback, see https://github.com/nodejs/node/pull/9652.)

> Is there are a debugger for the lib files themselves?

Chromium DevTools should work for them like it does for regular userland source files, I think?
